includes/quiz-submission/submission/repositories/class-submission-comments-based-repository.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * File containing the Submission_Comments_Based_Repository class.
+ *
+ * @package sensei
+ */
+
+namespace Sensei\Quiz_Submission\Submission\Repositories;
+
+use DateTime;
+use DateTimeInterface;
+use RuntimeException;
+use Sensei\Quiz_Submission\Submission\Models\Submission;
+use Sensei_Utils;
+use WP_Comment;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Submission_Comments_Based_Repository.
+ *
+ * @since $$next-version$$
+ */
+class Submission_Comments_Based_Repository implements Submission_Repository_Interface {
+	/**
+	 * Creates a new quiz submission.
+	 *
+	 * @param int        $quiz_id     The quiz ID.
+	 * @param int        $user_id     The user ID.
+	 * @param float|null $final_grade The final grade.
+	 *
+	 * @return Submission       The course progress.
+	 * @throws RuntimeException In case the lesson status is missing.
+	 */
+	public function create( int $quiz_id, int $user_id, float $final_grade = null ): Submission {
+		$status_comment = $this->get_status_comment( $quiz_id, $user_id );
+
+		if ( ! $status_comment ) {
+			throw new RuntimeException( 'Missing lesson status.' );
+		}
+
+		if ( null !== $final_grade ) {
+			update_comment_meta( $status_comment->comment_ID, 'grade', $final_grade );
+		}
+
+		$created_at = $this->get_created_date( $status_comment );
+
+		return new Submission(
+			$status_comment->comment_ID,
+			$quiz_id,
+			$user_id,
+			$final_grade,
+			$created_at
+		);
+	}
+
+	/**
+	 * Get or create a new quiz submission if it doesn't exist.
+	 *
+	 * @param int        $quiz_id     The quiz ID.
+	 * @param int        $user_id     The user ID.
+	 * @param float|null $final_grade The final grade.
+	 *
+	 * @return Submission The course progress.
+	 */
+	public function get_or_create( int $quiz_id, int $user_id, float $final_grade = null ): Submission {
+		$submission = $this->get( $quiz_id, $user_id );
+
+		if ( $submission ) {
+			return $submission;
+		}
+
+		return $this->create( $quiz_id, $user_id, $final_grade );
+	}
+
+	/**
+	 * Gets a course progress.
+	 *
+	 * @param int $quiz_id The quiz ID.
+	 * @param int $user_id The user ID.
+	 *
+	 * @return Submission|null The quiz submission.
+	 */
+	public function get( int $quiz_id, int $user_id ): ?Submission {
+		$status_comment = $this->get_status_comment( $quiz_id, $user_id );
+
+		if ( ! $status_comment ) {
+			return null;
+		}
+
+		$final_grade = get_comment_meta( $status_comment->comment_ID, 'grade', true );
+		$created_at  = $this->get_created_date( $status_comment );
+
+		return new Submission(
+			$status_comment->comment_ID,
+			$quiz_id,
+			$user_id,
+			$final_grade ? $final_grade : null,
+			$created_at
+		);
+	}
+
+	/**
+	 * Get the question IDs related to this quiz submission.
+	 *
+	 * @param int $quiz_id The quiz ID.
+	 * @param int $user_id The user ID.
+	 *
+	 * @return array An array of question post IDs.
+	 */
+	public function get_question_ids( int $quiz_id, int $user_id ): array {
```

Comment:
Does it make sense for this method to be part of the answer repository? 🤔

Reply:
If it extracts data from the submission then it's okay.